experimental/benchmark/src/database.rs

```diff
@@ -0,0 +1,123 @@
+//
+// Copyright 2020 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+use crate::proto::{Location, PointOfInterest};
+use log::info;
+use rand::Rng;
+use serde::{Deserialize, Serialize};
+use std::{collections::HashMap, time::Instant};
+
+/// Database structure represents internal XML fields from the following database:
+/// https://tfl.gov.uk/tfl/syndication/feeds/cycle-hire/livecyclehireupdates.xml
+#[derive(Debug, Default, Serialize, Deserialize, PartialEq)]
+struct StationDatabase {
+    #[serde(rename = "lastUpdate", default)]
+    last_update: String,
+    version: String,
+    #[serde(rename = "station", default)]
+    stations: Vec<Station>,
+}
+
+#[derive(Debug, Default, Serialize, Deserialize, PartialEq)]
+#[serde(rename = "station", default)]
+struct Station {
+    id: u32,
+    name: String,
+    #[serde(rename = "terminalName", default)]
+    terminal_name: String,
+    #[serde(rename = "lat", default)]
+    latitude_degrees: f32,
+    #[serde(rename = "long", default)]
+    longitude_degrees: f32,
+    #[serde(default)]
+    installed: bool,
+    #[serde(default)]
+    locked: bool,
+    #[serde(rename = "installDate", default)]
+    install_date: String,
+    #[serde(rename = "removalDate", default)]
+    removal_date: String,
+    temporary: bool,
+    #[serde(rename = "nbBikes", default)]
+    number_of_bikes: u32,
+    #[serde(rename = "nbEmptyDocks", default)]
+    number_of_empty_docks: u32,
+    #[serde(rename = "nbDocks", default)]
+    number_of_docks: u32,
+}
```

Comment:
If you are re-creating the example instead of using the existing one, why carry over all this complexity? Just create key / value blobs of pseudo-random data.

Reply:
I'm using the existing example, these fields are just for generating the data, that can (should be) parsed by the example.
I was planning to move this database to smth like `trusted_database/common`

Reply:
Then all this stuff should just be under `trusted_database/benchmark` I guess? if it's specific to that example anyways

Reply:
It's still experimental, so I thought I'll put it in `experimental` for now